src/main/java/uk/ac/ox/oxfish/fisher/equipment/fads/FadManagerUtils.java

```diff
@@ -31,23 +30,28 @@
 import java.util.stream.Stream;
 
 import static uk.ac.ox.oxfish.utility.MasonUtils.bagToStream;
-import static uk.ac.ox.oxfish.utility.MasonUtils.oneOf;
 
 /**
  * This provides convenience implementations for the various classes that need to access the
  * FadManager instance buried in the fisher's PurseSeineGear.
  */
 public interface FadManagerUtils {
 
+    static Stream<Fad> fadsHere(Fisher fisher) { return bagToStream(getFadManager(fisher).getFadsHere()); }
+
     static FadManager getFadManager(Fisher fisher) {
-        if (fisher.getGear() instanceof PurseSeineGear)
-            return ((PurseSeineGear) fisher.getGear()).getFadManager();
-        else throw new IllegalArgumentException(
+        return maybeGetFadManager(fisher).orElseThrow(() -> new IllegalArgumentException(
             "PurseSeineGear required to get FadManager instance. Fisher " +
                 fisher + " is using " + fisher.getGear().getClass() + "."
```

Comment:
will this work with decorators? 
Maybe not a big deal if it doesn't

Reply:
You mean with decorators of `PurseSeineGear`? I hadn't thought of that, but no, it wouldn't. If the need arises, I think we'd have to turn `PurseSeineGear` into an interface and have an implementation that can be decorated.